### PR TITLE
GH-835: Made sure forked process is terminated on application quit.

### LINKED ISSUE #835: [electron] Make sure all the forked processes terminate after the Electron application shutdown.
See here: https://github.com/theia-ide/yangster-electron/issues/17

### PR DESCRIPTION
**Pull request by kittaakos**
Adjusted server worker log messages, logged PID too.

Closes #835.

Signed-off-by: Akos Kitta <kittaakos@gmail.com>

### CODE REVIEW

File: dev-packages/application-package/src/generator/frontend-generator.ts [hunk 1]
```diff
@@ -120,6 +120,11 @@ if (cluster.isMaster) {
                 console.error(error);
                 electron.app.exit(1);
             });
+            electron.app.on('quit', function() {
+                // If we forked the process for the clusters, we need to manually terminate it.
+                // See: https://github.com/theia-ide/theia/issues/835
+                process.kill(cp.pid);
+            })
```
Comment (kittaakos) [inline dev-packages/application-package/src/generator/frontend-generator.ts]: I missed the semicolon here. :(

**Comment by kittaakos** [comment]
I have verified the changes on OS X with some trick from [here](https://github.com/theia-ide/yangster-electron/compare/GH-17). It fixes the issue.

The application works, but it does not fix anything on Windows. The LS process is a different store. I still need to verify one more use case on Ubuntu.

**Comment by kittaakos** [comment]
> store

I meant: "story".

**Comment by kittaakos** [comment]
Something is odd. I can select a WS root, but then I cannot change. I do not know whether it is related to my change or not. I will update this thread.

**Comment by kittaakos** [comment]
> Something is odd. I can select a WS root, but then I cannot change. I do not know whether it is related to my change or not. I will update this thread.

https://github.com/theia-ide/theia/issues/838

File: dev-packages/application-package/src/generator/frontend-generator.ts [hunk 2]
```diff
@@ -120,6 +120,11 @@ if (cluster.isMaster) {
                 console.error(error);
                 electron.app.exit(1);
             });
+            electron.app.on('quit', function() {
+                // If we forked the process for the clusters, we need to manually terminate it.
```
Comment (akosyakov) [inline dev-packages/application-package/src/generator/frontend-generator.ts]: I wonder is it the same when cluster forks, should we clean up for them too? Right now electron process -> master process -> worker process -> ls process. This change kills the master process. But if the worker process is still running, when the ls process won’t auto shutdown itself according the lsp.

Comment (kittaakos) [reply dev-packages/application-package/src/generator/frontend-generator.ts]: I think so. I had the same feeling. But I would like to tackle that as part of #815. I plan to proceed with smaller changes and verify them one by one on each platform with the bundled electron application.

Comment (akosyakov) [reply dev-packages/application-package/src/generator/frontend-generator.ts]: My point is that probably nothing should be done about ls servers, only workers should be killed, and ls processes will exit according to the lsp.

Comment (kittaakos) [reply dev-packages/application-package/src/generator/frontend-generator.ts]: OK. We'll see how that works in practice, especially on Windows in the context of #815.

**Pull request merged**